### ISSUE TITLE
Document all possible BugIds

### ISSUE THREAD

**Issue by SkyLined** [SkyLined/BugId#68]
I should create a definitive list and documentation for all possible BugIds.

**Comment by SkyLined** [comment]
Obviously, I need to find a way to make sure the list stays up-to-date; preferably by auto-generating it, but if that is not practically possible, by detecting when a BugId format changes, or a new class of BugId is created and the documentation has not been updated.

**Comment by JsHuang** [comment]
Have you finished the document?
When using BugId, I'm a little confused with some BugId meaning, eg. "**BOF**"?
And I want to know what's the meaning of undocumented part of BugId like **`[8]+0`** in id `BugId OOBR[8]+0 7b9.227`

**Comment by SkyLined** [comment]
No, unfortunately I have not.

BugId outputs an id but also a description (both on the command line and in the report). This description explains what the issue is in human readable form. The id is merely a very short version of this that uses abbreviations and symbols.

In the "**OOBR[8]+0**" case you will see something like this in the description: "*An Access Violation exception happened at <address> while attempting to read memory at <address>; at the end of a 8 bytes heap block at <address>. This indicates an Out-Of-Bounds (OOB) access bug was triggered.*"

This means the application attempted to read the first byte immediately following an 8 byte buffer. That means this is an Out-of-Bounds (**OOB**) Read (**R**) from an 8 bytes (**[8]**) buffer at offset 0 beyond the buffer (**+0**). The Bug id is therefore "**OOBR[8]+0**".

**BOF** stands for **B**uffer **O**ver**F**low; an attempt to write data into a buffer sequentially beyond the end of that buffer.
